### ISSUE TITLE
Error when using add_canonical_attributes on 0 dimensional data / Bug in cf_xarray.utils._is_datetime_like

### ISSUE THREAD

**Issue by rcaneill** [xarray-contrib/cf-xarray#285]
## Expected result
Using `da.cf.add_canonical_attributes()` on a 0 dimensional dataarray should return a new DataArray with the attributes added.

## What is happening
Due to the check on the data to check if this is a datetime, it raises an IndexError

## Reproducible code

```python
import xarray as xr
import cf_xarray
xr.DataArray(0, attrs={'standard_name':'sea_water_potential_temperature'}).cf.add_canonical_attributes()
```

Output:
```
IndexError                                Traceback (most recent call last)
/tmp/ipykernel_885592/3858241190.py in <module>
----> 1 xr.DataArray(0, attrs={'standard_name':'sea_water_potential_temperature'}).cf.add_canonical_attributes()

~/.local/share/virtualenvs/2021-11_obs_transition_zone-Sj4UnckL/lib/python3.8/site-packages/cf_xarray/accessor.py in add_canonical_attributes(self, override, skip, verbose, source)
   1802 
   1803                         # Don't add units to time variables (e.g., datetime64, ...)
-> 1804                         if key == "units" and _is_datetime_like(ds[var_name]):
   1805                             continue
   1806 

~/.local/share/virtualenvs/2021-11_obs_transition_zone-Sj4UnckL/lib/python3.8/site-packages/cf_xarray/utils.py in _is_datetime_like(da)
     19         import cftime
     20 
---> 21         if isinstance(da.data[0], cftime.datetime):
     22             return True
     23     except ImportError:

IndexError: too many indices for array: array is 0-dimensional, but 1 were indexed
```

Indeed, `_is_datetime_like` checks the `da.data[0]` which does not work for 0 dimension data (nor for dimension >= 2):
```python
import cftime
from cf_xarray.utils import _is_datetime_like
_is_datetime_like(xr.DataArray([[cftime.datetime(2022, 1, 12)]]))
```
Returns False.

`cf_xarray.__version__` is 0.6.2

## Proposed solution

I can submit a PR with a fix and tests. I guess replacing `if isinstance(da.data[0], cftime.datetime):` in `_is_datetime_like` by
`if isinstance(da.data.reshape(-1)[0], cftime.datetime):` would be sufficient.

**Comment by malmans2** [comment]
Sounds good to me!
Is reshape OK with all array types and quick?
Maybe better `isinstance(da.data[0] if da.shape else da.data, cftime.datetime)`?

**Comment by rcaneill** [comment]
It works with dask and numpy backends, and returns a view of the array (when possible) so I expect it to be fast
(see [numpy doc on ravel](https://numpy.org/doc/stable/reference/generated/numpy.ravel.html) in the notes "When a view is desired in as many cases as possible, arr.reshape(-1) may be preferable.").

`isinstance(da.data[0] if da.shape else da.data, cftime.datetime)` does not work for >= 2 dimension data I guess?

**Comment by dcherian** [comment]
:+1: A PR is very welcome!

**Comment by rcaneill** [comment]
Closed by PR #289
